crates/volta-core/src/manifest/serial.rs

```diff
@@ -177,6 +168,28 @@ impl Manifest {
     }
 }
 
+impl RawBinManifest {
+    pub fn into_bin_manifest(self) -> super::BinManifest {
```

Comment:
We could also do this with the `From` trait (which also gets us `.into()` for free), though presumably in the module with `BinManifest` rather than here:

```rust
impl From<RawBinManifest> for BinManifest {
    fn from(raw: RawBinManifest) -> Self {
        // exactly the same implementation
    }
}
```

Implementing `From` will give us `into` for free, and is also the [idiomatic] way to do this:

> Library authors should not directly implement this trait, but should prefer implementing the `From` trait, which offers greater flexibility and provides an equivalent Into implementation for free, thanks to a blanket implementation in the standard library.

[idiomatic]: https://doc.rust-lang.org/1.34.1/std/convert/trait.Into.html

The net is we'll be able to write this:

```rust
fn takes_bin_manifest(bin_manifest: BinManifest) {
    // ...
}

takes_bin_manifest(a_raw_bin_manifest.into());
```

Reply:
As a related note: 1.34 stabilized the `TryFrom` and `TryInto` traits, so a fair number of places where we have custom handling for `into_...` which is `Fallible`, we can convert to implementations of `TryFrom` instead. Definitely not for *this* PR; I'll open an issue tracking it, though.

Reply:
Good call, that is simpler! I opted to leave the `From` implementation in the `serial.rs` module, however, since it's related to the serialization, so it seems like it would fit better there.